components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1729,6 +1729,10 @@ void RewardsServiceImpl::SetContributionAmount(double amount) const {
   bat_ledger_->SetContributionAmount(amount);
 }
 
+void RewardsServiceImpl::SetContributionAmountForTesting(double amount) const {
```

Comment:
Do we need this extra function? Or is `SetContributionAmount` not accessible in test context or something

Reply:
I had set up friend classes before but tests would keep giving me private access only errors

Reply:
RewardsServiceImpl and BraveRewardsBrowserTest need to be in the same namespace

Reply:
Done. Needed to forward declare tests